ckanext/xloader/utils.py

```diff
@@ -107,6 +110,61 @@ def get_xloader_user_apitoken():
     return site_user["apikey"]
 
 
+def modify_ckan_url(result_url: str, ckan_url: str) -> str:
+    """ Modifies a URL based on CKAN site URL comparison.
+
+    This function compares the base URL of a given result URL against a CKAN site URL.
+    If they differ, the result URL is modified to use the CKAN site URL as its base
+    while preserving the original path.
+    
+    Args:
+        result_url (str): The original URL to potentially modify
+        ckan_url (str): The base CKAN site URL to compare against
+    Returns:
+        str: The modified URL if base URLs differ, otherwise returns original URL unchanged 
+    """
+    parsed_url = urlparse(result_url)
+    base_url = f"{parsed_url.scheme}://{parsed_url.netloc}"
+    path_url = parsed_url.path
+
+    if base_url != ckan_url:
+        result_url = urljoin(ckan_url, path_url)
+         
+    return result_url
+
+    
+def modify_resource_url(orig_ckan_url: str) -> str:
+    """Returns a potentially modified CKAN URL.
+
+    This function takes a CKAN URL and potentially modifies its base URL while preserving the path, 
+    query parameters, and fragments. The modification occurs only if two conditions are met:
+    1. The base URL of the input matches the configured CKAN site URL
+    2. An xloader_site_url is configured in the settings
+    
+    Args:
+        orig_ckan_url (str): The original CKAN URL to potentially modify
+    Returns:
+        str: Either the modified URL with new base URL from xloader_site_url, 
+             or the original URL if conditions aren't met
+    """
+    xloader_site_url = config.get('ckanext.xloader.site_url')
+    ckan_site_url = config.get('ckan.site_url')
+
+    parsed_url = urlparse(orig_ckan_url)
```

Comment:
This isn't needed until we're inside the conditional.

Reply:
Thanks @ThrawnCA  - Whereabouts? AFAICT I'm seeing `xloader_site_url`,  `ckan_site_url` and `parsed_url` needed for the first conditional ie: `if base_url == ckan_site_url and xloader_site_url:` (`parsed_url` is needed to determine `base_url`)

Reply:
Ah, I missed the usage inside the f-string.